Styling child components with Astro3 defaults (#4910)
withastro/docs: src/content/docs/en/guides/styling.mdx

In Astro, HTML attributes like `class` do not automatically pass through to child components.

> Instead, accept a `class` prop in the child component and apply it to the root element. When destructuring, you must rename it, because `class` is a [reserved word](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#reserved_words) in JavaScript. The default styles scoping strategy of Astro also requires you to pass the `data-astro-cid-*` attribute, and the simplest way to do this is by taking the `...rest` of the props and forwarding it to the component.

Reviewer: ```suggestion
Instead, accept a `class` prop in the child component and apply it to the root element. When destructuring, you must rename it, because `class` is a [reserved word](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#reserved_words) in JavaScript. 

When using the default `scopedStyleStrategy: 'attribute'`, you must also pass the `data-astro-cid-*` attribute. You can do this by also passing the `...rest` of the props to the component.
```
Just making some shorter, more active sentences!

Author: Right, I see what the goal is.

After looking at the suggested wording, I wrote a response and deleted it because I have not noticed the word "default". Is this also an indication of it not standing out enough? What do you think about this alternative?

> You must also pass the `data-astro-cid-*` attribute, unless you have changed `scopedStyleStrategy` to `class` or `where`. You can do this by passing the `...rest` of the props to the component.

Reviewer: Ooh, that makes a lot of sense, yes! Even better than the original, I think!

Author: Thank you! I have updated the PR to use this sentence, could you please review it again?

In Astro, HTML attributes like `class` do not automatically pass through to child components.

Instead, accept a `class` prop in the child component and apply it to the root element. When destructuring, you must rename it, because `class` is a [reserved word](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#reserved_words) in JavaScript.

> You must also pass the `data-astro-cid-*` attribute, unless you have changed `scopedStyleStrategy` to `'class'` or `'where'`. You can do this by passing the `...rest` of the props to the component.

Reviewer: ```suggestion
Using the default scoped style strategy, you must also pass the `data-astro-cid-*` attribute. You can do this by passing the `...rest` of the props to the component. If you have changed `scopedStyleStrategy` to `'class'` or `'where'`, the `...rest` prop is not necessary.
```
Just separating more clearly the two cases here, instead of mixing the "unless" in the middle of the need and the way to achieve it.

(This also means we can remove line 164, I think?)

<MyComponent class="red">This will be red!</MyComponent>
```

> This pattern lets you style child components directly. If your [`scopedStyleStrategy` option](/en/reference/configuration-reference/#scopedstylestrategy) is set to either `'where'` or `'class'` you can remove the `...rest` props to make it simpler.

Reviewer: ```suggestion
This pattern lets you style child components directly. If your [`scopedStyleStrategy` option](/en/reference/configuration-reference/#scopedstylestrategy) is set to either `'where'` or `'class'`, the `...rest` props are not necessary.
```

<MyComponent class="red">This will be red!</MyComponent>
```

> This pattern lets you style child components directly. If your [`scopedStyleStrategy` option](/en/reference/configuration-reference/#scopedstylestrategy) is set to either `'where'` or `'class'`, the `...rest` props are not necessary.

Author: Yeah, I noticed this duplication earlier but you found a way to avoid it altogether :+1:

> This pattern lets you style child components directly. If your [`scopedStyleStrategy` option](/en/reference/configuration-reference/#scopedstylestrategy) is set to either `'where'` or `'class'`, the `...rest` props are not necessary.

Reviewer: Avoiding writing is what I do best! lol. Let's ship this!

> This pattern lets you style child components directly. If your [`scopedStyleStrategy` option](/en/reference/configuration-reference/#scopedstylestrategy) is set to either `'where'` or `'class'`, the `...rest` props are not necessary.

Author: This is the way :joy:
